similar/main/ai.cpp

```diff
@@ -89,7 +89,7 @@ using std::min;
 #define	BABY_SPIDER_ID	14
 
 namespace dsx {
-static void init_boss_segments(boss_special_segment_array_t &segptr, int size_check, int one_wall_hack);
+static void init_boss_segments(objptridx_t boss_objnum, boss_special_segment_array_t &segptr, int size_check, int one_wall_hack);
```

Comment:
Since the first step in the called function is to reject `object_none`, you could use `vobjptridx_t` to enforce that no caller can pass `object_none`.